server/src/main/java/org/apache/iotdb/db/utils/TypeInferenceUtils.java

```diff
@@ -63,6 +69,9 @@ public static TSDataType getPredictedDataType(Object value, boolean inferType) {
         return booleanStringInferType;
       } else if (isNumber(strValue)){
         if (!strValue.contains(TsFileConstant.PATH_SEPARATOR)) {
+          if (isConvertFloatPrecisionLack(strValue)) {
```

Comment:
how about just check the length of the string?

Reply:
this will only happen at the first insertion of a time series, don't worry